src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -183,35 +186,36 @@ def frobenius_endomorphism(self):
         deg = self._base.over(Fq).degree(Fq)
         return self._Hom_(self, category=self.category())(t**deg)
 
-    def frobenius_charpoly(self, var='X'):
+    def frobenius_charpoly(self, var='X', algorithm='crystalline'):
         r"""
         Return the characteristic polynomial of the Frobenius
-        endomorphism if the rank is two. Raise a NotImplementedError
-        otherwise.
+        endomorphism.
 
         Let `\mathbb{F}_q` be the base field of the function ring. The
-        *characteristic polynomial `\chi` of the Frobenius endomorphism*
+        *characteristic polynomial* `\chi` *of the Frobenius endomorphism*
         is defined in [Gek1991]_. An important feature of this
         polynomial is that it is a monic univariate polynomial with
         coefficients in the function ring. As in our case the function
         ring is a univariate polynomial ring, it is customary to see the
         characteristic polynomial of the Frobenius endomorphism as a
         bivariate polynomial.
 
-        Let `\chi = X^2 - A(T)X + B(T)` be the characteristic polynomial
-        of the Frobenius endomorphism, and let `t^n` be the Ore polynomial
-        that defines the Frobenius endomorphism of `\phi`; by
-        definition, `n` is the degree over of the base field over
-        `\mathbb{F}_q`. We have `\chi(t^n)(\phi(T)) = t^{2n} - \phi_A
-        t^n + \phi_B = 0`, with `\deg(A) \leq \frac{n}{2}` and `\deg(B)
-        = n`.
+        Let `\chi = X^r + \sum_{i=1}^{r} A_{i}(T)X^{r-i}` be the
+        characteristic polynomial of the Frobenius endomorphism, and
+        let `t^n` be the Ore polynomial that defines the Frobenius
+        endomorphism of `\phi`; by definition, `n` is the degree of `K`
+        over the base field `\mathbb{F}_q`. Then we have `\chi(t^n)(\phi(T))
+        = t^{nr} + \sum_{i=1}^{r} \phi_{A_{i}}t^{n(r-i)} = 0`, with `\deg(A_i)
+        \leq \frac{ir}{n}`.
 
-        Note that the *Frobenius trace* is defined as `A(T)` and the
-        *Frobenius norm* is defined as `B(T)`.
+        Note that the *Frobenius trace* is defined as `A_1(T)` and the
+        *Frobenius norm* is defined as `A_r(T)`.
 
         INPUT:
 
         - ``var`` (default: ``'X'``) -- the name of the second variable
+        - ``algorithm`` (default: ``'crystalline'``) -- the algorithm used to compute
+            the characteristic polynomial
```

Comment:
PEP8: docstring and comment lines should be less than 72 characters long.

```suggestion
        INPUT:

        - ``var`` (default: ``'X'``) -- the name of the second variable
        - ``algorithm`` (default: ``'crystalline'``) -- the algorithm
          used to compute the characteristic polynomial
```